pkg/decode/decode_protobuf.go

```diff
@@ -65,16 +62,20 @@ func PBFlowToMap(flow *pbflow.Record) config.GenericMap {
 	if flow.Packets != 0 {
 		out["Packets"] = flow.Packets
 	}
-	var interfaces []interface{}
-	var flowDirections []interface{}
 
 	if len(flow.GetDupList()) != 0 {
+		var interfaces []interface{}
+		var directions []interface{}
 		for _, entry := range flow.GetDupList() {
 			interfaces = append(interfaces, entry.Interface)
-			flowDirections = append(flowDirections, entry.Direction)
+			directions = append(directions, entry.Direction)
 		}
 		out["Interfaces"] = interfaces
-		out["FlowDirections"] = flowDirections
+		out["FlowDirections"] = directions
+	} else {
+		out["Interface"] = flow.Interface
+		out["FlowDirection"] = int(flow.Direction.Number())
```

Comment:
can we here just append flow.Interface to interfaces and the same with direction and just have plural in the cm ?

Reply:
That's an approach yes but it means you will also have array if we decide to keep the `justMark` option.